GitHubCore.ps1

```diff
@@ -1087,6 +1087,10 @@ filter ConvertTo-SmarterObject
             {
                 $property.Value = ConvertTo-SmarterObject -InputObject $property.Value
             }
+
+            # Add the property to the clone.
+            $propertyClone = New-Object -TypeName PSNoteProperty -ArgumentList $property.Name, $property.Value
```

Comment:
This approach would introduce a regression in that `$InputObject.$property` would be getting modified _in addition_ to `$clone.$property`.  The intent of this was to provide a no-sideeffect method that could create a "smart" object without making any changes to the original object.

Reply:
Then I'll remove that last commit and just go with the drop-in replacement for `BinaryFormatter`.